Assets/MRTK/SDK/Features/UX/Scripts/BoundsControl/Visuals/PerAxisHandles.cs

```diff
@@ -78,14 +78,14 @@ private void UpdateColliderType()
                 if (oldBoxCollider != null && config.HandlePrefabColliderType == HandlePrefabCollider.Sphere)
                 {
                     shouldCreateNewCollider = true;
-                    Object.Destroy(oldBoxCollider);
+                    Object.DestroyImmediate(oldBoxCollider);
```

Comment:
Ahh I see your explanation now:

> A call to Destroy() is replaced with DestroyImmediate(), as it was causing issues with sub-frame component checks for the presence of colliders. If Destroy() is called, but then the code checks for the presence of the Destroy()ed collider within the same frame, the check will incorrectly pass.

How do the checks happen? Is it from a `GetComponent` or a raycast or something? I feel like there should be something we could set to null or check an additional flag instead of using `DestroyImmediate`.

Even Unity's docs _very_ strongly dissuade from using DestroyImmediate at runtime:

> Destroys the object obj immediately. You are strongly recommended to use Destroy instead.
> This function should only be used when writing editor code since the delayed destruction will never be invoked in edit mode. In game code you should use Object.Destroy instead.

Reply:
The frame delay does cause issues. The way it manifests is that later code checks for the presence of the collider with `EnsureComponent`. The problem with this is that if a delayed `Destroy` has been initiated, `EnsureComponent` will think that the collider is actually there, return a reference to that collider, but then break a frame later because the collider it returned was actually scheduled for destruction.

```
public static T EnsureComponent<T>(this GameObject gameObject) where T : Component
        {
            T foundComponent = gameObject.GetComponent<T>();
            return foundComponent == null ? gameObject.AddComponent<T>() : foundComponent;
        }
```

This is the relevant code that checks for the presence of the component. It is "fooled" by a component that is scheduled for destruction by `Destroy`. `AddComponentsToAffordance` is what calls `EnsureComponent`.

Reply:
@keveleigh Switched to `Destroy()`, but added warning comments for future trespassers :)

Reply:
Sounds good! Thank you!